inginious/frontend/pages/course_user_settings.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+"""
+Module definition for CourseUserSettingPage class
+
+This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+more information about the licensing of this file.
+"""
+
+import flask
+from inginious.common.field_types import FieldTypes
+from inginious.frontend.pages.utils import INGIniousAuthPage
+
+
+class CourseUserSettingPage(INGIniousAuthPage):
+    """
+    Class definition for CourseUserSettingPage
+    """
+
+    def GET_AUTH(self, courseid):
+        """ GET request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        current_user = self.database.users.find_one(
+            {"username": username})
+        course_user_settings = current_user.get("course_settings", {})
+        return self.show_page(courseid, course_user_settings.get(courseid, {}), None)
+
+    def POST_AUTH(self, courseid):
+        """ POST request """
+        username = self.user_manager.session_username()
+        if not self._is_accessible(courseid, username):
+            return self.template_helper.render("course_unavailable.html")
+        course_user_settings = self._sanitize_content(flask.request.form, courseid)
+        if type(course_user_settings)==str:
```

Comment:
This is not very Pythonic as you perform a try/except in `_sanitize_content`  that returns a string in the except block, and then compare for the type here. You can simply put the try/except block here around the `_sanitize_content` call, so that the method has clean specs : returns the data, and throws an exception in case of errors.